http/api.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+
+	. "github.com/StevensSEC/pokemonbattlelib"
+)
+
+func buildRouter() *http.ServeMux {
+	mux := http.NewServeMux()
+	mux.HandleFunc("/pokedex/list/moves", HandleListMoves)
+	mux.HandleFunc("/pokedex/generate", HandleGeneratePokemon)
+	mux.HandleFunc("/battle/new", HandleCreateBattle)
+	mux.HandleFunc("/battle/simulate", HandleBattleSimulate)
+	mux.HandleFunc("/agent/dumb", HandleDumbAgent)
+	return mux
+}
```

Comment:
I've pretty much exclusively used Gin for web serving in Go. Just handles a lot more out of the box if you want something more sophisticated.

Reply:
I'm trying to avoid adding dependencies as much as possible.